Sources/SPTDataLoader.m

```diff
@@ -240,10 +289,19 @@ - (void)receivedInitialResponse:(SPTDataLoaderResponse *)response
     }
     
     if ([self.delegate respondsToSelector:@selector(dataLoader:didReceiveInitialResponse:)]) {
-        [self executeDelegateBlock: ^{
+        [self executeDelegateBlock:^{
             [self.delegate dataLoader:self didReceiveInitialResponse:response];
         }];
     }
+    [self executeDelegateBlock:^{
+        void (^handler)(SPTDataLoaderResult *) = [self.requestHandlers objectForKey:response.request];
+        if (handler) {
+            handler([SPTDataLoaderResult fromResponse:response kind:SPTDataLoaderResultReceivedInitialResponse]);
+            @synchronized(self.requestHandlers) {
+                [self.requestHandlers removeObjectForKey:response.request];
```

Comment:
I still don't get why this is being done. You are attaching a single callback block but are firing it on _any_ delegate method but you are also removing the callback every time after it is called. None of the subsequent events will fire in the block as you have removed the callback. Only the first one received will ever be fired.

This API also feels sort of clunky and weird (one callback block that is fired many times for many different types of events whether you want them or not). What is the purpose of this?

Reply:
If you're going to use a callback for this, it should only be removed for terminal events (request is completely finished).

The naming convention you've gone for is also nonintuitive. The block is called `completion` but it's not really a "completion" block if it's called many times for every type of delegated event. It's almost more of a subscription with how it seems you've defined this API.